script/publish.ts

```diff
@@ -124,36 +128,39 @@ function upload(assetName, assetPath) {
     Bucket: bucket,
     ACL: 'public-read',
     Key: key,
-    Body: fs.createReadStream(assetPath),
+    Body: Fs.createReadStream(assetPath),
   }
 
-  return new Promise((resolve, reject) => {
-    s3.upload(uploadParams, (error, data) => {
-      if (error) {
-        reject(error)
-      } else {
-        const stats = fs.statSync(assetPath)
-        const hash = crypto.createHash('sha1')
-        const input = fs.createReadStream(assetPath)
-
-        hash.on('finish', () => {
-          const sha = hash.read().toString('hex')
-          resolve({ name: assetName, url, size: stats['size'], sha })
-        })
-
-        input.pipe(hash)
+  return new Promise<IUploadResult>((resolve, reject) => {
+    s3.upload(
+      uploadParams,
+      (error: Error, data: AWS.S3.ManagedUpload.SendData) => {
+        if (error) {
+          reject(error)
+        } else {
+          const stats = Fs.statSync(assetPath)
+          const hash = Crypto.createHash('sha1')
+          const input = Fs.createReadStream(assetPath)
+
+          hash.on('finish', () => {
+            const sha = hash.read().toString('hex')
```

Comment:
TS *really* doesn’t like this. Should this be `hash.digest().toString('hex')`?

Reply:
`digest` computes based on the current content, but I still think we want to `read` here and process some data. We've assumed here that `read()` returns a `Buffer`, but we don't check for `string` here. That's what TS is complaining about.

TL;DR: if we get a string here, I'm not sure what we should do about it. Let's throw here and figure it out later:

```ts
hash.on('finish', () => {
  const result = hash.read()
  if (result instanceof Buffer) {
    const sha = result.toString('hex')
    resolve({ name: assetName, url, size: stats['size'], sha })
  } else {
    reject('hash.read() returned a string?!?!?!?')
  }
})
```

Reply:
Is the latest commit better?

Reply:
Yeah, that seems better.